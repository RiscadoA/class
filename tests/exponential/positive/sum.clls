proc main() {
    cut {
        println("!"); !a(b: choice of { |#C: close |#W: wait });
        println("#A"); #C b;
        println("close"); close b
        |a: ? offer of { |#C: wait |#W: close }|
        println("?"); ?a;
        println("call 1"); call a(y);
        println("case of 1"); case y of {
            |#C:
                println("wait 1"); wait y;
                println("call 2"); call a(z);
                println("case of 2"); case z of {
                    |#C:
                        println("wait 2"); wait z;
                        ()
                    |#W:
                        println("never"); close z
                }
            |#W:
                println("never"); close y

        }
    }
};;
