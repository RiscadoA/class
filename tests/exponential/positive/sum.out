!
#A
close
?
call 1
case of 1
wait 1
call 2
case of 2
wait 2
